src/hypercorn/protocol/quic.py

```diff
@@ -93,11 +103,49 @@ async def handle(self, event: Event) -> None:
                 and header.packet_type == PACKET_TYPE_INITIAL
                 and not self.context.terminated.is_set()
             ):
+                cid = header.destination_cid
+                retry_cid = None
+                if self.retry is not None:
+                    if not header.token:
+                        if header.version is None:
+                            return
```

Comment:
Why return here, is a missing version an indication of an error?

Reply:
Retry is an option because it trades off increased latency for ensuring that you can roundtrip to the client and aren't just being used to amplify an attack (with the start of the TLS handshake) by someone spoofing UDP.  I usually turn it on.

Re the missing version... if there is no header token, then this isn't the client retrying, and we have to ask them to retry, but to do that we need the version.  If there isn't a version, they've sent us a versionless "short header" packet which is not a sensible thing to do, so we just drop the packet.